deblur/workflow.py

```diff
@@ -61,17 +62,25 @@ def dereplicate_seqs(seqs_fp,
     min_size : integer, optional
         discard sequences with an abundance value smaller
         than integer
-    uc_output: boolean, optional
-        output the dereplication map in .uc format
+    use_log: boolean, optional
+        save the vsearch logfile as well (to output_fp.log)
+        default=False
     """
-    log_name = "%s.log" % splitext(output_fp)[0]
+    logger = logging.getLogger(__name__)
+    logger.debug('dereplicate seqs file %s' % seqs_fp)
+
+    log_name = "%s.log" % output_fp
 
-    vsearch_dereplicate_exact_seqs(
-        fasta_filepath=seqs_fp,
-        output_filepath=output_fp,
-        output_uc=uc_output,
-        minuniquesize=min_size,
-        log_name=log_name)
+    params = ['vsearch', '--derep_fulllength', seqs_fp, '--output', output_fp, '--sizeout']
+    params.extend(['--fasta_width', '0', '--threads', '1', '--minuniquesize', str(min_size)])
+    params.extend(['--quiet'])
+    if use_log:
+        params.extend(['--log', log_name])
+    res = subprocess.call(params)
```

Comment:
Instead of using subprocess call, can you use this function from qiita? https://github.com/biocore/qiita/blob/master/qiita_db/processing_job.py#L25

You can just copy/paste that function and use it here directly (it is BSD licensed so no need for extra permission as long as we put a note saying that we copied form Qiita). The advantages of it is that it returns the std_err and std_out, which you can include in the logger.debug/logger.error calls which will help for debugging.